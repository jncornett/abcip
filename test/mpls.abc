d ( stack="eth:mpls:ip4:tcp" )
a ( syn )
a ( syn, 3:lse=0x12345F78 )
a ( syn, 3:lab=0x12340, tcl=0x5, bos=0x1, ttl=0x08 )

d ( stack="eth:mpls:ip6:tcp" )
a ( syn )

d ( stack="eth:mpls:eth:ip4:tcp" )
a ( syn )

d ( stack="eth:mpls:mpls:mpls:mpls:ip4:tcp" )
a ( syn, 5:lse=0x50000000 )
