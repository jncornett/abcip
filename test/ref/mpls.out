MPLS (label 0 (IPv4 explicit NULL), exp 0, [S], ttl 64)
	IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.4.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d8 (correct), seq 1, win 8192, length 0[|MPLS]
	0x0000:  0209 0807 0605 0201 0203 0405 8847 0000  .............G..
	0x0010:  0140 4500 0028 0001 0000 4006 5cbb 0a01  .@E..(....@.\...
	0x0020:  0204 0a09 0807 bdec 0008 0000 0001 0000  ................
	0x0030:  0000 5002 2000 b3d8 0000                 ..P.......
MPLS (label 74565, exp 7, [S], ttl 120)
	IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.4.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d8 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8847 1234  .............G.4
	0x0010:  5f78 4500 0028 0002 0000 4006 5cba 0a01  _xE..(....@.\...
	0x0020:  0204 0a09 0807 bdec 0008 0000 0001 0000  ................
	0x0030:  0000 5002 2000 b3d8 0000                 ..P.......
MPLS (label 74560, exp 5, [S], ttl 8)
	IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.4.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d8 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 8847 1234  .............G.4
	0x0010:  0b08 4500 0028 0003 0000 4006 5cb9 0a01  ..E..(....@.\...
	0x0020:  0204 0a09 0807 bdec 0008 0000 0001 0000  ................
	0x0030:  0000 5002 2000 b3d8 0000                 ..P.......
MPLS (label 0 (IPv4 explicit NULL), exp 0, [S], ttl 64)
	IP6 , wrong link-layer encapsulationbad-hlen 0[|MPLS]
	0x0000:  0209 0807 0605 0201 0203 0405 8847 0000  .............G..
	0x0010:  0140 6000 0001 0014 0640 0000 0000 0000  .@`......@......
	0x0020:  0000 0000 ffff 0a01 0204 0000 0000 0000  ................
	0x0030:  0000 0000 ffff 0a09 0807 bdec 0008 0000  ................
	0x0040:  0001 0000 0000 5002 2000 b3d8 0000       ......P.......
MPLS (label 0 (IPv4 explicit NULL), exp 0, [S], ttl 64)
	IP0 bad-hlen 8[|MPLS]
	0x0000:  0209 0807 0605 0201 0203 0405 8847 0000  .............G..
	0x0010:  0140 0209 0807 0605 0201 0203 0405 0800  .@..............
	0x0020:  4500 0028 0001 0000 4006 5cba 0a01 0205  E..(....@.\.....
	0x0030:  0a09 0807 bdec 0008 0000 0001 0000 0000  ................
	0x0040:  5002 2000 b3d7 0000                      P.......
MPLS (label 0 (IPv4 explicit NULL), exp 0, ttl 64)
	(label 0 (IPv4 explicit NULL), exp 0, ttl 64)
	(label 327680, exp 0, ttl 0)
	(label 0 (IPv4 explicit NULL), exp 0, [S], ttl 64)
	IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.7.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d5 (correct), seq 1, win 8192, length 0[|MPLS]
	0x0000:  0209 0807 0605 0201 0203 0405 8847 0000  .............G..
	0x0010:  0040 0000 0040 5000 0000 0000 0140 4500  .@...@P......@E.
	0x0020:  0028 0001 0000 4006 5cb8 0a01 0207 0a09  .(....@.\.......
	0x0030:  0807 bdec 0008 0000 0001 0000 0000 5002  ..............P.
	0x0040:  2000 b3d5 0000                           ......
